smile_id_core/web_api.py

```diff
@@ -106,6 +109,34 @@ def verify_document(
         if last_name:
             payload["last_name"] = last_name
 
+        return payload
+
+    def verify_document_async(
+        self,
+        **data
+    ):
+        """
+            Performs a document verification request with an immediate response in async way.
+            **data consists of parameters for _prepare_document_verification_payload() method above to create payload
+            for request.
+
+            :return: dict
+        """
+        payload = self._prepare_document_verification_payload(**data)
+        response = self._make_request("POST", self.Urls.VERIFY_DOCUMENT_ASYNC, payload, api_version=2)
```

Comment:
The API v1 also have async doc verification, so I would suggest using `self.api_version` here so the method can be used in both api v1 and api v2 initialized clients

Reply:
I have an idea, maybe if these clients are so semantically different, instead of passing the api version to the constructor create a class AsyncAPI, and while maintaining the same public interface, just override the make_request method. THat seems to be more rational, more DRY and less confusing

Reply:
well, it is not exactly Async api, but rather next version of it :thinking: I do like a separate class since we discovered that urls are actually different as well, would make things simpler for sure